src/coreclr/jit/codegenxarch.cpp

```diff
@@ -11470,6 +11449,45 @@ void CodeGen::genRestoreCalleeSavedFltRegs(unsigned lclFrameSize)
     }
 }
 
+//-----------------------------------------------------------------------------------
+// genClearAvxStateInProlog: Generate vzeroupper instruction to clear AVX state if necessary in a prolog
+//
+void CodeGen::genClearAvxStateInProlog()
+{
+    if (GetEmitter()->ContainsCallNeedingVzeroupper() && !GetEmitter()->Contains256bitOrMoreAVX())
```

Comment:
Just double checking, the funclet calls will return `true` for `ContainsCallNeedingVzeroupper`, correct?

Reply:
Yes, the flag is for the whole method. It is approximation. We will emit vzeroupper even when there is no actual call needing it in the given funclet.